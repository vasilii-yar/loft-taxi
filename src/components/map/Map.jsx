import React from 'react';
import NavBar from "../navbar/NavBar";
import MapBox from "./MapBox";
import {connect} from "react-redux";
import RegistrationRedirectForm from "./RegistrationRedirectForm";
import RouteForm from "./RouteForm";
import {fetchProfileData} from "../../redux/modules/profile/profileActions";

class Map extends React.Component {

    componentDidMount() {
        this.props.fetchProfileData();
    }

    render() {
        const Form = this.props.isCardDataFilled ? RouteForm : RegistrationRedirectForm
        return (
            <>
                <header>
                    <NavBar/>
                </header>
                <main>
                    <section>
                        <MapBox/>
                        <Form/>
                    </section>
                </main>
            </>
        );
    }
}

const mapStateToProps = (state) => {
    return (
        {
            isCardDataFilled: state.profile.isCardDataFilled
        }
    )
}

export default connect(
    mapStateToProps,
    {fetchProfileData}
    )(Map);