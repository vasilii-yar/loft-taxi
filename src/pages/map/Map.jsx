import React from 'react';
import NavBar from "../../components/navbar/NavBar";
import MapBox from "../../components/mapbox/MapBox";

export default class Map extends React.Component {
    render() {
        return (
            <>
                <header>
                    <NavBar/>
                </header>
                <main>
                    <section>
                        <MapBox/>
                    </section>
                </main>
            </>
        );
    }
}