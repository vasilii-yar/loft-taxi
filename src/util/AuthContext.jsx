import React, {useEffect} from 'react';

export const Context = React.createContext();

/*export default class AuthContext extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isLoggedIn: false
        };
    }

    logIn = (email, password) => {
        this.setState({isLoggedIn: true});
    }

    logOut = () => {
        this.setState({isLoggedIn: false});
    }

    isLoggedIn = () => {
        return this.state.isLoggedIn;
    }

    render() {
        return (
            <Context.Provider value={{
                logIn: this.logIn,
                logOut: this.logOut,
                isLoggedIn: this.isLoggedIn
            }}>
                {this.props.children}
            </Context.Provider>
        );
    }
}*/

export const AuthProvider = ({children}) => {
    const [isLoggedIn, setIsLoggedIn] = React.useState(false);

    const logIn = (email, password) => {
        setIsLoggedIn(true);
    }

    const logOut = () => {
        setIsLoggedIn(false);
    }

    return (
        <Context.Provider value={{logIn, logOut, isLoggedIn}}>
            {children}
        </Context.Provider>
    );
}

export const withAuth = (WrappedComponent) => {
    return class extends React.Component {
        render() {
            return (
                <Context.Consumer>
                    {
                        (value) => {
                            return <WrappedComponent {...value} {...this.props}/>
                        }
                    }
                </Context.Consumer>
            );
        }
    }
}
