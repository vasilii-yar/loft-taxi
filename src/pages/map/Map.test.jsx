import React from 'react';
import {render} from "@testing-library/react";
import Map from "./Map";
import {AuthProvider} from "../../util/AuthContext";
import renderer from "react-test-renderer";

const mockNavigateTo = jest.fn();
jest.mock("../../components/mapbox/MapBox", () => (
    {
        __esModule: true,
        default: () => <div>Карта</div>
    }
));
jest.mock("../../components/navbar/NavBar", () => (
    {
        __esModule: true,
        default: () => <div>NavBar component</div>
    }
));
jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
    Map: () => ({})
}));

describe("Map", () => {
    it("renders correctly", () => {
        const {container} = render(
            <AuthProvider>
                <Map navigateTo={mockNavigateTo}/>
            </AuthProvider>
        );
        expect(container.innerHTML).toMatch("Карта");
    })

    it("renders correctly check by snapshot", () => {
        const tree = renderer.create(
            <AuthProvider>
                <Map navigateTo={mockNavigateTo}/>
            </AuthProvider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
});