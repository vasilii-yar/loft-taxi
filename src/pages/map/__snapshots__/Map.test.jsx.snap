// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map renders correctly check by snapshot 1`] = `
Array [
  <header>
    <div>
      NavBar component
    </div>
  </header>,
  <main>
    <section>
      <div>
        Карта
      </div>
    </section>
  </main>,
]
`;
